Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -141,7 +141,7 @@ HTMLInputElement::~HTMLInputElement()
     // a radio button that was in a form. The call to setForm(nullptr) above
     // actually adds the button to the document groups in the latter case.
     // That is inelegant, but harmless since we remove it here.
-    if (isRadioButton())
+    if (m_inputType && isRadioButton())
```

Comment:
What is this for? Change log doesn’t say.

Reply:
I'll add in the commit log. A while back, we landed a Speedometer optimization in HTMLInputElement which delays initialization of the inputType when constructed by the parser. This was to avoid creating an inputType for "text" only to figure out later during parsing that we wanted another type.

There is a case in the new HTML fast parsing case where we give up parsing a destroy a newly constructed HTMLInputElement before we're done fully parsing it. As a result, the input element may now get destroyed *before* its m_inputType got initialized. However, the HTMLInputElement would previously crash if m_inputType is null. It seemed like a waste to force the initialization of m_inputType in this case so I decided to go the other way and add support for an HTMLInputElement to get destroyed before its m_inputType gets initialized.